kornia/augmentation/random_generator/_2d/random_rain.py

```diff
@@ -74,3 +78,65 @@ def forward(self, batch_shape: tuple[int, ...], same_on_batch: bool = False) ->
             'drop_height_factor': drop_height_factor,
             'drop_width_factor': drop_width_factor,
         }
+
+    def random_rain_augmentation(
+        self,
+        image_val: torch.Tensor,
+        num_raindrops: int = 100,
+        min_length: int = 5,
+        max_length: int = 15,
+        color: Optional[torch.Tensor] = None,
+    ) -> torch.Tensor:
+        """Apply random rain augmentation to the input image.
+        Args:
+            image (torch.Tensor): the input image with shape :math:`(C,H,W)`.
+            num_raindrops (int, optional): number of raindrops to draw. Defaults to 100.
+            min_length (int, optional): minimum length of a raindrop. Defaults to 5.
+            max_length (int, optional): maximum length of a raindrop. Defaults to 15.
+            color (torch.Tensor, optional): the color of the raindrops. Defaults to torch.tensor([255]).
+
+        Returns:
+            torch.Tensor: The image with raindrops.
+        Doctests:
```

Comment:
```suggestion
        Examples:
```
i suggest to run `make doctest` before commit to make sure this will pass the ci

Reply:
@edgarriba, should it be a test class? Doctests are written and successfully run locally.

Reply:
yes, test class is a must. The docstring we use only to showcase in the docs which is also tested in the ci